spec/requests/api/v1/user_trails/create_spec.rb

```diff
@@ -0,0 +1,92 @@
+      )
+      county = County.create!(
+        id: 27,
+        name: "Boulder"
+      )
+      trail = Trail.create!(
+        id: 163,
+        name: "Jenny Creek Trail ",
+        latitude: "39.929561",
+        longitude: "-105.624647",
+        difficulty: "black",
+        distance: "2.2",
+        description: "Jenny Creek Trail is a lesser-used trail that goes from Yankee Doodle Lake Eldora Ski area. In it motorized 2 track and in summer is an out-and-back from Rollins pass road or Jenny Creek Road with no summer access through Eldora.",
+        start_elevation: "10,732 ft",
+        avg_duration: "0:25:11",
+        map_image: "https://ep1.pinkbike.org/trailstaticmap/563000/563447_3_500x200.png",
+        thumbnail_image: "https://adventr.co/wp-content/uploads/2015/01/DSC03243.jpg",
+        county_id: 27
+      )
+
+      headers = { "Content-Type": "application/json", Accept: "application/json" }
+      body = ( {
+        "user_id": 3,
+        "trail_id": 163
+      } )
+
+      expect(shawn.trails.count).to eq(0)
+
+      post '/api/v1/user-trails', headers: headers, params: JSON.generate(body)
+      
+      expect(response).to be_successful
+      expect(response.status).to eq(200)
+      
+      parsed = JSON.parse(response.body, symbolize_names: true)
+            
+      expect(parsed).to be_a(Hash)
+      expect(parsed).to have_key(:success)
+      expect(parsed[:success]).to eq("Trail Added Successfully")
+
+      expect(shawn.trails.count).to eq(1)
+    end
+  end
+
+  describe 'when user id is not valid' do
+    it 'returns an error message' do
+      county = County.create!(
+        id: 27,
+        name: "Boulder"
+      )
+      trail = Trail.create!(
+        id: 163,
+        name: "Jenny Creek Trail ",
+        latitude: "39.929561",
+        longitude: "-105.624647",
+        difficulty: "black",
+        distance: "2.2",
+        description: "Jenny Creek Trail is a lesser-used trail that goes from Yankee Doodle Lake Eldora Ski area. In it motorized 2 track and in summer is an out-and-back from Rollins pass road or Jenny Creek Road with no summer access through Eldora.",
+        start_elevation: "10,732 ft",
+        avg_duration: "0:25:11",
+        map_image: "https://ep1.pinkbike.org/trailstaticmap/563000/563447_3_500x200.png",
+        thumbnail_image: "https://adventr.co/wp-content/uploads/2015/01/DSC03243.jpg",
+        county_id: 27
+      )
+
+      headers = { "Content-Type": "application/json", Accept: "application/json" }
+      body = ( {
+        "user_id":  nil,
+        "trail_id": 163
+      } )
+
+      post '/api/v1/user-trails', headers: headers, params: JSON.generate(body)
+      
+      expect(response).to be_successful
+      expect(response.status).to eq(200)
+
+      parsed = JSON.parse(response.body, symbolize_names: true)
+      
+      expect(parsed).to be_a(Hash)
+      expect(parsed).to have_key(:error)
+      expect(parsed[:error]).to eq("No User Found")
+    end
+  end
+end
```

Comment:
I would love to see another test case for when there is missing data (i.e. a trail id is wrong or missing)